frontend/src/breeding_form.tsx

```diff
@@ -116,22 +129,311 @@ export function BreedingForm({
+        "warning"
+      );
+      return false;
+    }
+
+    if (userInput.breed_date !== null && userInput.birth_date !== null) {
+      const datesValid = validateDates(
+        userInput.breed_date,
+        userInput.birth_date
+      );
+      if (datesValid === false) {
+        return false;
+      }
+    }
+    return true;
+  };
+
+  const updateBreeding = async (breedingUpdates: Breeding) => {
+    // Only run this if there is a breeding to update.
+    if (data == "new") {
+      return;
+    }
+
+    const currentHerdBreedings = await get(`/api/breeding/${herdId}`);
+
+    const breedingMatch = currentHerdBreedings.breedings.find(
+      (item) =>
+        item.mother == data.mother &&
+        item.father == data.father &&
+        (item.breed_date == data.breed_date ||
+          item.birth_date == data.birth_date)
+    );
+
+    if (!breedingMatch) {
+      userMessage("Parningstillfället kunde inte hittas.", "error");
+      return;
+    }
+
+    const modifyBreedingUpdates = (updates: Breeding) => {
+      let newUpdates: Breeding = { ...updates, ...{ id: breedingMatch.id } };
+      for (let key in newUpdates) {
+        if (newUpdates[key] === null || newUpdates[key] === undefined) {
+          delete newUpdates[key];
+        }
+      }
+      return newUpdates;
+    };
+
+    const modifiedBreedingUpdates = modifyBreedingUpdates(breedingUpdates);
+
+    const breedingUpdateResponse = await patch(
+      "/api/breeding",
+      modifiedBreedingUpdates
+    );
+
+    if (breedingUpdateResponse.status === "success") {
+      return breedingUpdateResponse;
+    }
+
+    const translate: Map<string, string> = new Map([
+      ["Not logged in", "Du är inte inloggad. Logga in och försök igen"],
+      [
+        "Unknown mother",
+        "Okänd mor, modern måste vara en aktiv individ i databasen",
+      ],
+      [
+        "Unknown father",
+        "Okänd far, fadern måste vara en aktiv individ i databasen",
+      ],
+      [
+        "Unknown mother, Unknown father",
+        "Okända föräldrar. Både modern och fadern måste vara aktiva individer i databasen.",
+      ],
+      ["Forbidden", "Du har inte behörighet att skapa parningstillfället."],
+    ]);
+
+    if (
+      breedingUpdateResponse.status == "error" &&
+      !!breedingUpdateResponse.message &&
+      translate.has(breedingUpdateResponse.message)
+    ) {
+      userMessage(translate.get(breedingUpdateResponse.message), "error");
+      return;
+    }
+
+    userMessage(
+      "Okänt fel - något gick fel på grund av tekniska problem. Kontakta en administratör.",
+      "error"
+    );
+    return;
+  };
+
+  const createBreeding = async (breedingData: LimitedBreeding) => {
+    const breedingCreationResponse: {
+      status: string;
+      message?: string;
+      breeding_id?: number;
+    } = await post("/api/breeding", breedingData);
+
+    if (breedingCreationResponse.status == "success") {
+      return breedingCreationResponse;
+    }
```

Comment:
this looks like a good workaround as long as get don't get proper http codes back